src/model/TaskManager.cpp

```diff
@@ -46,25 +46,35 @@ std::optional<Core::TaskID> TaskManager::ParentOf(const Core::TaskID &id) const
         return std::nullopt;
 }
 
-ActionResult TaskManager::Add(const Core::Task &t) {
-    Core::TaskID id = gen_->genID();
-    if (IsPresent(id))
-        return {ActionResult::Status::DUPLICATE_ID, id};
+Core::ModelInquiryResult TaskManager::Add(const Core::Task &t) {
+    Core::ModelInquiryResult result;
+    auto id = gen_->genID();
 
-    tasks_.insert(std::make_pair(id, std::make_pair(t, Node())));
-    return {ActionResult::Status::SUCCESS, id};
+    if (ToBool(IsPresent(id))) {
+        result.set_status(Core::ModelInquiryResult_Status_DUPLICATE_ID);
```

Comment:
Case when `IdGenerator` has created duplicate id is internal error of `TaskManager`, why should user of `TaskManager` API know about it? He didn't enter any id's that could be duplicate.

Reply:
Fair point, I removed this check form Add and AddSubtask (also tests)